src/app/app-routing.module.ts

```diff
@@ -1,14 +1,21 @@
 import { NgModule } from '@angular/core';
 import { Routes, RouterModule } from '@angular/router';
 
-import { CoursesComponent } from './courses/courses.component';
 import { LoginPageComponent } from './auth/login-page/login-page.component';
+import { NoMatchPageComponent } from './core/no-match-page/no-match-page.component';
+import { AuthGuard } from './auth/auth.guard';
+import { CoursesModule } from './courses/courses.module';
 
 const routes: Routes = [
   { path: 'auth', component: LoginPageComponent },
-  { path: 'courses', component: CoursesComponent },
-  { path: '', redirectTo: '/auth', pathMatch: 'full' },
-  { path: '**', component: CoursesComponent },
+  {
+    path: 'courses',
+    loadChildren: () => CoursesModule,
```

Comment:
It's not lazy loading. Please, read about it and open your network tab to check your module is loaded as a separate junk.

Reply:
corrected